### PR TITLE
Copy editorconfig and pre-commit hook from Cardano

### PR DESCRIPTION
**Pull request by michaelpj**
- Editorconfig is nice for making sure your editor picks up the indentation settings etc.
    - I removed the makefile settings because seriously, who uses make?
- The pre-commit hook is handy now we have stylish and hlint on the CI.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
```
Comment (vmchale) [inline .editorconfig]: We use 4 spaces for indentation.

Comment (michaelpj) [reply .editorconfig]: scroll down for the `.hs` section :)

Comment (michaelpj) [reply .editorconfig]: I think in practice this means it applies to the nix code.

Comment (vmchale) [reply .editorconfig]: Ah okay.

**Pull request merged**